### PR TITLE
Use terraform package for check-divergence

### PR DESCRIPTION
**Pull request by poornima-krishnasamy**
This PR uses terraform package to check divergence of terraform code. This also removed redundant plan and apply commands.

### CODE REVIEW

File: pkg/terraform/terraform.go [hunk 1]
```diff
@@ -127,7 +121,10 @@ func NewTerraformCLI(config *TerraformCLIConfig) (*TerraformCLI, error) {
 // `terraform workspace new <name>`
 func (t *TerraformCLI) Init(ctx context.Context) error {
 	var wsCreated bool
+	var out bytes.Buffer
 
+	t.tf.SetStdout(&out)
```
Comment (jasonBirchall) [inline pkg/terraform/terraform.go:126]: I'm not sure about forcing this on the caller. Should it be an argument? 
`func (t *TerraformCLI) Init(ctx context.Context, w writer) error {` etc

Comment (jasonBirchall) [reply pkg/terraform/terraform.go:126]: and then you can `t.tf.SetStderr(&w)` or something similar.

Comment (jasonBirchall) [reply pkg/terraform/terraform.go:126]: Wasn't there a cool method that let you specify your stdout at one point?

Comment (poornima-krishnasamy) [reply pkg/terraform/terraform.go]: @jasonBirchall , I have updated to pass the Write as an argument but still prints the output using the terraform method. let me know if that looks ok

**Review by jasonBirchall — APPROVED** [review]

**Pull request merged**